app/models/mapping_rule.rb

```diff
@@ -20,6 +20,26 @@ def as_json(mapping_rules, version:)
       }
     end
 
+    def as_txt(mappings, version:)
+      header = <<~HEADER
+        # Shopify Product Taxonomy - Mappings: #{version}
+        # Format:
+        # input_taxonomy: <input taxonomy version>
+        # output_taxonomy: <output taxonomy version>
+        # {full_name} → {full_name}
+      HEADER
+      sorted_mapping = mappings.sort_by do |mapping|
+        mapping.input.full_name.nil? ? 0 : -mapping.input.full_name.size
+      end
+      padding = sorted_mapping.first.input.full_name.size
```

Comment:
Why in memory vs a query (like we do with category for example: `padding = reorder("LENGTH(id) desc").first.gid.size`)

Reply:
I guess here you're dealing with stronger subsets so you have to rely on the in-memory passed values...

Reply:
Yea, from an aesthetic perspective, we prefer not to have an unusually long category name from a channel taxonomy distort the formatting for other mappings.

Reply:
Yep very aligned